rfcs/git-sdk-requirements/RFC.md

```diff
@@ -0,0 +1,697 @@
+
+> on second review this idea might not make sense as perhaps part of 'git thesis'
+
+> is access to the fs itself, not fetched through git sdk interface
+
+> however if that interface is indistinguishable from the file system but with git powers attached
+
+> why pass some fs and complicate setup potentially
+
+## Git compiled to WASM
+
+### Context
+
+The reason you'd want to move to libgit2 is that libgit2 supports more options out of the box. And potentially performance.
+
+### Questions
+
+#### Q: Is git compiled to WASM faster than JS implementation?
+
+Most likely yes as WASM in general is faster. Compilation is done by [Emscripten](https://emscripten.org/).
+
+It takes the C code and creates a `.wasm` file.
+
+libgit2 was not written with WASM in mind but it still will most likely be faster than isomorphic git.
+
+The resulting `.wasm` file in `Release` mode is ~ 830 KB.
+
+#### Q: What are missing APIs Git SDK will need. Does libgit2 provide them?
+
+libgit2 is written in C. So changes to it must be made in C too if libgit2 doesn't support a certain feature.
+
+Fortunately it seems libgit2 supports all the needed features Inlang needs. And potentially other features that other companies building on Git SDK will need.
+
+From list shared above with checkboxes for available API. API of libgit2 is [here](https://libgit2.org/libgit2/#HEAD).
+
+```js
+// elementary
+clone() // ✅ https://libgit2.org/docs/guides/101-samples/#repositories_clone_simple
+shallowClone() // ✅ part of clone, there is open pr for it that builds, tested it works!
+commit() // ✅ https://libgit2.org/docs/guides/101-samples/#commits
+push() // ✅ https://stackoverflow.com/questions/28055919/how-to-push-with-libgit2
+pull() // ✅ https://stackoverflow.com/questions/39651287/doing-a-git-pull-with-libgit2
+sparseCheckout() // ❓ https://github.com/libgit2/libgit2/issues/2263 (open pr, need to build, test)
+
+// branch related
+currentBranch() // ✅ https://stackoverflow.com/questions/12132862/how-do-i-get-the-name-of-the-current-branch-in-libgit2
+createBranch() // ✅ https://libgit2.org/libgit2/#HEAD/group/branch/git_branch_create
+renameBranch() // ✅ https://libgit2.org/libgit2/#v0.17.0/group/branch
+switchBranch() // ✅ https://stackoverflow.com/questions/46757991/checkout-branch-with-libgit2
+deleteBranch() // ✅ https://libgit2.org/libgit2/#v0.17.0/group/branch/git_branch_delete
+
+// change related
+// (3 API different "changes" concepts...)
+unstagedChanges() // ❓ didn't find example with this, but it may exist
+uncommittedChanges() // ❓ didn't find example too
+unpushedChanges() // ✅ https://stackoverflow.com/questions/42131934/get-unpushed-commits-with-libgit2
+
+// host like GitHub or GitLab dependent
+// ouside of libgit2 scope
+// would need to be built as part of Git SDK and send relevant commands to libgit2
+signIn()
+signOut()
+createFork()
+syncFork()
+openPullRequest()
+```
+
+#### Q: Is 833 KB WASM file an issue to load?
+
+Should be non issue as its under 1 MB.
```

Comment:
What makes you believe that? 

Assuming that SSR is important, what implications will an 800KB WASM file have on SEO? Will a JS implementation avoid large bundle sizes?